src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -2063,6 +2063,45 @@ closeGraphContext(unsigned int graphId)
   return ret ? -errno : ret;
 }
 
+void
+shim::
+open_graph_context(const zynqaie::hwctx_object* hwctx, const uuid_t xclbinId, unsigned int graph_id, xrt::graph::access_mode am)
+{
+  if (!hw_context_enable){
+    // for legacy flow
+    if (auto ret = openGraphContext(xclbinId, graph_id, am))
+      throw xrt_core::error("Failed to open graph context");
+    return;
+  }
```

Comment:
Spacing, and if you don't use the `ret` value then don't save it.

```suggestion
  if (!hw_context_enable) {
    // for legacy flow
    if (openGraphContext(xclbinId, graph_id, am))
      throw xrt_core::error("Failed to open graph context");
      
    return;
  }
```

Reply:
Hi @stsoe I've pushed the changes. Requesting to review.